x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/InternalUsers.java

```diff
@@ -122,6 +125,35 @@ public class InternalUsers {
         )
     );
 
+    /**
+     * Internal user that manages DLM. Has all indices permissions to perform DLM runtime tasks.
+     */
+    public static final InternalUser DLM_USER = new InternalUser(
+        UsernamesField.DLM_NAME,
+        new RoleDescriptor(
+            UsernamesField.DLM_ROLE,
+            new String[] {},
+            new RoleDescriptor.IndicesPrivileges[] {
+                RoleDescriptor.IndicesPrivileges.builder()
+                    // There are no active plans to manage the security index or async search index with DLM, so excluding them here
+                    .indices("/@&~(\\.security.*)&~(\\.async-search.*)/")
+                    .privileges(
+                        "delete_index",
+                        RolloverAction.NAME,
```

Comment:
I've gone with an explicit list here of actions here (instead of something like `manage` or `maintenance`) to keep privileges as paired down as possible. Let me know if you disagree.